docs/developer-guide/cstor-pool-docs.md

```diff
@@ -0,0 +1,189 @@
+cStor pool cluster can have minimum 1 cStor pool in order to finally create a volume. Users can decide how many
+cStor pools they want to be there in a cStor pool cluster.
+
+To create cStor pool cluster, CSPC custom resource (API) can be used. One need to put the details in a CSPC YAML file, for example nodes and associated 
+disks they want to select with other relevant information and then do a `kubectl apply -f <cspc-yaml-file>` to create a cStor pool cluster.
+To learn more about how to create pool cluster using CSPC, follow this link. (TODO)
+
+
+*CSPC is a declarative API where you could specify the intent/configuration of your pool cluster to be configured in a Kubernetes 
+cluster and cStor components would converge the system towards the specified intent.* 
+
+## CStor pool Custom Resource and Component
+
+### CSPC-Operator Introduction
+
+This sections describes the custom resources and components involved in cStor pools.
+
+```
+                                                              +-------------------------+
+                                                              |                         |
+                                                              |                         |
+                                                              |                         |
+                                                              |                         |
+                                                              |      CSPC-Operator      |
+                                                              |                         |
+                               XXXXXXXXXXXXXXX                |                         |
+                              X              X                |                         |
+Admin/User/SRE               X               X                |                         |
+                            X                X  Reconcile     |                         |
+     XXX                   X                 +--------------->+                         |
+    XXXXX                  X                 X                |                         |
+      X                    X   CSPC-YAML     X                |                         |
+    XXXXX                  X                 X                |                         |
+      X        Applies     X                 X                +-------------------------+
+      X     +------------->X                 X
+     XXX                   X                 X
+    X X X                  X                 X
+      X                    X                 X
+                           X                 X
+                           XXXXXXXXXXXXXXXXXXX
+
+```
+
+1. CSPC contains a list of pool specs and a pool spec contains configuration for cStor pool creation on a particular node as specified in the pool spec.
+For example, if a CSPC manifest has 3 pool specs, 3 cStor pools will be created on 3 different nodes.
```

Comment:
I find this and line and 64 confusing together, maybe we can have one point just saying if we cannot have different pools on the same node .. I think that is the intention here ...

Reply:
Will do that. 
For max_pools -- In quick.md -- it is about replicas that can be placed on a CSPC. So the replicas should be less than or equal to the number of cStor pools in the CSPC. This will be mentioned in volume doc.